pkg/backoff/backoff.go

```diff
@@ -79,35 +72,35 @@ func WithBackoff(config Config) heartbeat.HandleOption {
 	}
 }
 
-// shouldBackoff returns true if the backoff should be applied. It also returns a boolean value
-// indicating whether the backoff should be reset.
-func shouldBackoff(retries int, at time.Time) (bool, bool) {
+// shouldBackoff returns true if we should save heartbeats directly to offline
+// database and skip sending to API due to rate limiting from too many recent
+// networking errors.
+func shouldBackoff(retries int, at time.Time) bool {
 	if retries < 1 || at.IsZero() {
-		return false, true
+		return false
 	}
 
-	next := float64(factor) * math.Pow(2, float64(retries))
-	if next > float64(resetAfter) {
+	seconds := int(float64(factor) * math.Pow(2, float64(retries)))
+
+	if seconds > int(maxBackoff.Seconds()) {
 		log.Debugf(
-			"exponential backoff tried %d times since %s, it will reset due to %d seconds limit",
+			"exponential backoff tried %d times since %s, will reset because reached %d hr max backoff",
```

Comment:
Seconds instead of hard-coded "hour"?
```suggestion
			"exponential backoff tried %d times since %s, will reset because reached %d secs max backoff",
```

Reply:
Actually, we should use a helper function to format the `time.Duration` in human readable form, or just use the build-in formatter:

https://stackoverflow.com/questions/41335155/time-duration-to-string-2h-instead-2h0m0s